include/chashmap/chashmap.h

```diff
@@ -0,0 +1,82 @@
+#ifndef CHASHMAP_H
+#define CHASHMAP_H
+
+#define DEFAULT_SEGMENT_LENGTH 100
+
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct HashmapEntry {
+    char* key;
+    char* value;
+
+    /* Pointer to next element */
+    void* next;
+} HashmapEntry;
+
+typedef struct Hashmap {
+    /* Amount of elements in HashmapEntry*/
+    long segment_length;
+
+    /* Actual array of hashmap entries */
+    void** HashmapEntries;
+
+    /* Byte size */
+    size_t size;
+
+    /* Amount of elements */
+    size_t length;

```

Comment:
I did not understand why there are 3 properties `segment_length`, `size` and `length`. Maybe it is enough to have size (number of entries) and capacity (size of allocated memory)